src/Microsoft.Extensions.Caching.Memory/MemoryCache.cs

```diff
@@ -272,6 +321,53 @@ private static void ScanForExpiredItems(MemoryCache cache)
             }
         }
 
+        private bool UpdateCacheSizeExceedsCapacity(CacheEntry entry, out bool cacheSizeUpdated)
+        {
+            if (!_options.SizeLimit.HasValue)
+            {
+                cacheSizeUpdated = false;
+                return false;
+            }
+
+            var newSize = 0L;
+            while(true)
```

Comment:
finite retries? e.g. if it fails 100 times then give up and return that you're over capacity. Under stress you may get a bunch of threads in this loop competing with eachother and there's no gurantee how long they'll be spinning. Some poor threads could be stuck here forever burning all the CPU on your machine (unlikely but possible). MemoryCache needs to be fast, and that includes failing fast if necessary.